pyartcd/pyartcd/jenkins.py

```diff
@@ -0,0 +1,67 @@
+import base64
+import logging
+import os
+from urllib.parse import urlencode
+
+from pyartcd import exectools
+
+logger = logging.getLogger(__name__)
+
+
+async def trigger_jenkins_job(job_path: str, params=None):
+    """
+    Trigger a job using remote API calls.
+    :param job_path: relative path to the job, starting from <buildvm hostname>:<jenkins port>
+    :param params: optional dict containing job parameters
+    """
+
+    service_account = os.getenv('JENKINS_SERVICE_ACCOUNT')
+    token = os.getenv('JENKINS_SERVICE_ACCOUNT_TOKEN')
+
+    if not service_account or not token:
+        logger.warning(
+            'Env vars JENKINS_SERVICE_ACCOUNT and JENKINS_SERVICE_ACCOUNT_TOKEN '
+            'must be defined in order to trigger jobs using remote API')
+        raise RuntimeError
+
+    # We need to use cURL --resolve option to pin the request to local host
+    # Doing so in pure Python requires a lot of hacking (see for example http://bit.ly/3ICezXw)
+    auth_header = base64.b64encode(f'{service_account}:{token}'.encode()).decode()
+    logger.info(f'auth_header: {auth_header}')
+
+    cmd = f'curl -X POST -H "Authorization: Basic {auth_header}" ' \
+          f'--resolve buildvm.hosts.prod.psi.bos.redhat.com:8443:127.0.0.1 ' \
+          f'https://buildvm.hosts.prod.psi.bos.redhat.com:8443/{job_path}'
```

Comment:
As a workaround, we can either adding `127.0.0.1 saml.buildvm.hosts.prod.psi.bos.redhat.com` to `/etc/hosts` to make it resolvable, or have a custom resolver if using `aiohttp`. But I am ok with this if this is simpler.

Reply:
Done